java-checks/src/main/java/org/sonar/java/checks/RedundantThrowsDeclarationCheck.java

```diff
@@ -272,6 +286,31 @@ public void visitClass(ClassTree tree) {
     public void visitLambdaExpression(LambdaExpressionTree lambdaExpressionTree) {
       // skip lambdas
     }
+
+    private static Optional<Symbol.MethodSymbol> getImplicitlyCalledConstructor(MethodTree methodTree) {
+      Type superType = ((Symbol.TypeSymbol)methodTree.symbol().owner()).superClass();
+      return superType.symbol().memberSymbols().stream()
+        .filter(MethodInvocationVisitor::isDefaultConstructor)
+        .map(Symbol.MethodSymbol.class::cast)
+        .findFirst();
+    }
+
+    private static boolean isDefaultConstructor(Symbol symbol) {
+      if (symbol.isMethodSymbol()) {
+        Symbol.MethodSymbol methodSymbol = ((Symbol.MethodSymbol)symbol);
+        if ("<init>".equals(methodSymbol.name())) {
```

Comment:
extract `"<init>"` into a constant, such as `CONSTRUCTOR_NAME`, to make it explicit and help future reviewer/reader to get the meaning.